sklearn/_display_estimator.py

```diff
@@ -0,0 +1,308 @@
+        estimators.
+        If type == 'single', then `estimators` is a single estimator.
+
+    names : list of str
+        If type in ('parallel', 'serial'), then `names` corresponds to
+        estimators
+        If type is 'single', then `names` is a single string corresponding to
+        the single estimator.
+
+    name_details : list of str, str, or None, default=None
+        If type == 'parallel', then `name_details` corresponds to `names`.
+        If type == 'single', then `name_details` is a single string
+        corresponding to the single estimator.
+        `name_details` is not used when type == 'single'.
+
+    dash_wrapped : bool, default=True
+        If true, wrapped HTML element will be wrapped with a dashed boarder.
+    """
+    def __init__(self, type, estimators, names, name_details=None,
+                 dash_wrapped=True):
+        self.type = type
+        self.estimators = estimators
+        self.names = names
+        self.dash_wrapped = dash_wrapped
+
+        if self.type == 'parallel' and name_details is None:
+            name_details = (None, ) * len(names)
+
+        self.name_details = name_details
+
+
+def _type_of_html_estimator(estimator):
+    """Generate information about how to display an estimator.
+    """
+    if isinstance(estimator, _EstHTMLBlock):
+        return estimator
+    elif isinstance(estimator, str):
+        return _EstHTMLBlock('single', estimator, estimator, estimator)
+    elif estimator is None:
+        return _EstHTMLBlock('single', estimator, 'None', 'None')
+    # looks like a meta estimator
+    elif (hasattr(estimator, 'estimator') and
+            hasattr(getattr(estimator, 'estimator'), 'get_params')):
+        wrapped_estimator = getattr(estimator, 'estimator')
+        wrapped_name = wrapped_estimator.__class__.__name__
+        return _EstHTMLBlock('serial', [wrapped_estimator], [wrapped_name])
+    return estimator._sk_repr_html()
+
+
+def _write_label_html(out, name, name_details,
+                      outer_class="sk-label-container",
+                      inner_class="sk-label",
+                      checked=False):
+    """Write labeled html with or without a dropdown with named details"""
+    out.write(f'<div class="{outer_class}">'
+              f'<div class="{inner_class} sk-toggleable">')
+
+    if name_details is not None:
+        checked_str = 'checked' if checked else ''
+        est_id = uuid.uuid4()
+        out.write(f'<input class="sk-toggleable__control sk-hidden--visually" '
+                  f'id="{est_id}" type="checkbox" {checked_str}>'
+                  f'<label class="sk-toggleable__label" for="{est_id}">'
+                  f'{name}</label>'
+                  f'<div class="sk-toggleable__content"><pre>{name_details}'
+                  f'</pre></div>')
+    else:
+        out.write(f'<label>{name}</label>')
+    out.write('</div></div>')  # outer_class inner_class
+
+
+def _write_named_label_html(out, estimator, name):
```

Comment:
If the purpose of this is to extract the name_details from the estimator, should we just permit `name_details=SomeEstimator()` in `_write_label_html` and get rid of this function?